### PR TITLE
Improve consistency and naming in user services

### PR DESCRIPTION
**Pull request by krusche**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.

### Motivation and Context
Some left-over improvements of previous changes in the user service

### Steps for Testing
1. Log in to Artemis (TS1, TS2)
2. Register as new user, activate account, forget password, change settings (TS2)
3. Create, update and delete users (TS1, TS2)

### Test Coverage
Nothing changed

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java [hunk 1]
```diff
@@ -258,6 +255,11 @@ default User getUserWithGroupsAndAuthorities(@NotNull String username) {
         return findOneWithGroupsAndAuthoritiesByLogin(login);
     }
 
+    @NotNull
+    default User findWithGroupsAndAuthoritiesByIdElseThrow(long userId) {
```
Comment (LDAP) [inline src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java:259]: In regard to consistency I would propose `findOneWith...`

File: src/main/java/de/tum/in/www1/artemis/service/user/UserCreationService.java [hunk 2]
```diff
@@ -127,7 +96,8 @@ public User createUser(String login, @Nullable String password, Set<String> grou
         newUser.setLogin(login);
         newUser.setFirstName(firstName);
         newUser.setLastName(lastName);
-        newUser.setGroups(groups);
+        // needs to be mutable --> new HashSet<>(Set.of())
+        newUser.setGroups(groups != null ? groups : new HashSet<>(Set.of()));
```
Comment (Strohgelaender) [inline src/main/java/de/tum/in/www1/artemis/service/user/UserCreationService.java:100]:
```suggestion
        newUser.setGroups(groups != null ? new HashSet<>(groups) : new HashSet<>());
```
This makes sure that the Set is always mutable. Also Set.of() is not needed to create an empty Set.

File: src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java (hunk 1 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/repository/UserRepository.java:259]: We actually would like to unify this with other repositories, so it should be `findByIdWith...ElseThrow`

All the methods with `findOne...` should be renamed accordingly

**Review by airelawaleria — APPROVED** [review]
tested in the testing session on ts1 (will test later on ts2 when it will be free). worked as expected

**Review by JohannesWeiss — APPROVED** [review]

**Review by EneaGore — APPROVED** [review]
Tested successfully on ts1.

**Review by simonzilker — APPROVED** [review]
What @airelawaleria said :)

**Review by marvinchimeka — APPROVED** [review]
Functionalities on ts1 worked fine. Test later in ts2. Tested in testing session

**Comment by krusche** [comment]
> Tested successfully on ts1.
> 
> new user registration didn't seem to work for me on ts2, but maybe it is an anomaly
> I received the error:
> Failed to load resource: the server responded with a status of 400 ()
> image

User Registration should work on TS2, please double check it

**Review by philgabr — APPROVED** [review]
Successfully tested on TS2, I did not encounter any issues :+1:

**Review by valeriebucher — APPROVED** [review]
Tested on ts2, everything worked!

**Pull request merged**